### PR TITLE
chore(clients/go): Refactor entities package with format and style updates

### PR DESCRIPTION
**Pull request by slewiskelly**
## Description

The following (non-behavioural) changes have been made:

 - Add package comment
 - Add comments for exported types and functions
 - Shorten variable names for those with limited scope
 - Display function and inputs in tests
 - Display diffs in tests
 - Add [github.com/google/go-cmp/cmp](https://godoc.org/github.com/google/go-cmp/cmp) to vendor directory
 - Add [github.com/google/go-cmp/cmp/cmptops](https://godoc.org/github.com/google/go-cmp/cmp/cmpopts) to vendor directory

The above changes have been made based on:
 - https://github.com/golang/go/wiki/CodeReviewComments
 - https://github.com/golang/go/wiki/TestComments

## Related issues

#3004 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: clients/go/entities/job_test.go [hunk 1]
```diff
@@ -15,72 +15,81 @@
...
 	}
-	if !reflect.DeepEqual(variables, expectedJsonAsMapVariables) {
-		t.Error("Failed to get variables as map, got", variables, "instead of", expectedJsonAsMap)
+
+	want := map[string]interface{}{
+		"foo":   "bar",
+		"hello": "world",
+	}
+
+	if diff := cmp.Diff(want, got, cmpopts.SortMaps(func(a, b string) bool { return a < b })); diff != "" {
```
Comment (MiguelPires) [inline clients/go/entities/job_test.go]: Why is `cmpopts.SortMaps` necessary?

File: clients/go/entities/job_test.go [hunk 2]
```diff
@@ -15,72 +15,81 @@
...
-		Variables:     expectedJson,
-	}}
 )
 
 func TestJob_GetVariablesAsMap(t *testing.T) {
-	variables, err := job.GetVariablesAsMap()
+	got, err := job.GetVariablesAsMap()
 	if err != nil {
-		t.Error("Failed to get variables as map", err)
+		t.Errorf("job.GetVariablesAsMap() = %v", err)
```
Comment (MiguelPires) [inline clients/go/entities/job_test.go]: I understand preferring `Error` to `Fail` but in this case `got` is invalid so we can fail here.

**Review by miguelpires — CHANGES_REQUESTED** [review]
Thank you for your contribution @slewiskelly, we really appreciate your help in improving the client's code quality. If you'd like to keep contributing, feel free to assign your PRs to me. 

I asked a couple of questions below

File: clients/go/entities/job_test.go (hunk 1 above)
Comment (slewiskelly) [reply clients/go/entities/job_test.go]: Normally Go maps don't guarantee any ordering. So for the sake of comparison, required sorting by key.

However, I didn't look into the implementation of `cmp.Equal` to see that the maps are sorted anyway: https://github.com/google/go-cmp/blob/b1c9c4891a6525d98001fea424c8926c6d77bb56/cmp/compare.go#L478.

File: clients/go/entities/job_test.go (hunk 2 above)
Comment (slewiskelly) [reply clients/go/entities/job_test.go]: Good point.

**Review by miguelpires — APPROVED** [review]
bors r+

Thanks for the contribution @slewiskelly

**Pull request merged**